### DISCUSSION TITLE
Swelling, thickness change, and active material volume fraction

### DISCUSSION THREAD

**Discussion by DavidMStraub** [pybamm-team/PyBaMM#2576] [Q&A]
Hi,

I have a question about the current particle mechanics submodels. If I understand correctly, they require a function parameter `{Domain} electrode volume change`, which is a function of stoichiometry and maximum concentration, which corresponds to the *particle* volume change. It is then assumed that the relative particle volume change translates 1:1 in a cell thickness change, since the yz-dimensions obviously don't change. However, it is currently not possible to model a change in electrode active material volume fraction caused by swelling, correct? Naively, one could think that swelling would lead to both active material volume fraction increase (i.e. porosity reduction) and thickness change and it would be interesting if that can be modelled.

**Comment by DavidMStraub** [comment]
I guess the answer is that this would require a mechanics submodel and a porosity submodel which are coupled, correct?

**Reply by valentinsulzer** [reply]
Yes, the porosity submodel could depend on the mechanics. It's not obvious whether active material volume fraction should increase though (since that is used to determine total lithium)

**Comment by DavidMStraub** [comment]
Oh, because active material volume fraction change would correspond to LAM. Got it. Will accept my own answer then 😉
